### ISSUE TITLE
Checkpoint silently not correctly restored.

### ISSUE THREAD

**Issue by phuntast1c** [Lightning-AI/pytorch-lightning#20089]
### Bug description

Continue a training run from a checkpoint does not correctly set the loaded checkpoint to the `_loaded_checkpoint` attribute of the `lightning.pytorch.trainer.connectors.checkpoint_connector._CheckpointConnector` object, thus not correctly restoring the checkpoint.
The bug can be fixed by replacing the line
        `self._loaded_checkpoint_loaded_checkpoint = _pl_migrate_checkpoint(loaded_checkpoint, checkpoint_path)`
with
        `self._loaded_checkpoint = _pl_migrate_checkpoint(loaded_checkpoint, checkpoint_path)`
in <class 'lightning.pytorch.trainer.connectors.checkpoint_connector._CheckpointConnector‘>.

### What version are you seeing the problem on?

v2.0.6

### Environment

<details>
  <summary>Current environment</summary>

* CUDA:
        - GPU:
                - NVIDIA RTX A5000
                - NVIDIA RTX A5000
                - NVIDIA RTX A5000
                - NVIDIA RTX A5000
        - available:         True
        - version:           11.8
* Lightning:
        - gradnorm-pytorch:  0.0.26
        - lightning:         2.0.6
        - lightning-cloud:   0.5.37
        - lightning-utilities: 0.8.0
        - pytorch-lightning: 2.0.0
        - torch:             2.0.1
        - torch-fftconv:     0.1.3
        - torchaudio:        2.0.2
        - torchinfo:         1.7.2
        - torchmetrics:      0.11.4
        - torchvision:       0.15.2
* Packages:
        - accelerate:        0.29.3
        - aiohttp:           3.8.4
        - aiosignal:         1.3.1
        - antlr4-python3-runtime: 4.8
        - anyio:             3.6.2
        - appdirs:           1.4.4
        - arrow:             1.2.3
        - asttokens:         2.0.5
        - async-timeout:     4.0.2
        - attrs:             22.2.0
        - audioread:         3.0.0
        - backcall:          0.2.0
        - backoff:           2.2.1
        - beartype:          0.18.5
        - beautifulsoup4:    4.12.0
        - black:             22.6.0
        - blessed:           1.20.0
        - bottleneck:        1.3.5
        - brotlipy:          0.7.0
        - certifi:           2023.7.22
        - cffi:              1.15.1
        - cftime:            1.6.2
        - charset-normalizer: 2.0.4
        - click:             8.0.4
        - cloudpickle:       2.2.1
        - coloredlogs:       15.0.1
        - contextlib2:       21.6.0
        - contourpy:         1.0.5
        - croniter:          1.3.8
        - cryptography:      39.0.1
        - cycler:            0.11.0
        - cython:            3.0.10
        - dateutils:         0.6.12
        - debugpy:           1.5.1
        - decorator:         5.1.1
        - deepdiff:          6.3.0
        - deprecated:        1.2.14
        - dnspython:         2.3.0
        - docker-pycreds:    0.4.0
        - docstring-parser:  0.15
        - einops:            0.7.0
        - email-validator:   1.3.1
        - executing:         0.8.3
        - fastapi:           0.101.0
        - filelock:          3.9.0
        - flatbuffers:       23.3.3
        - fonttools:         4.25.0
        - frozenlist:        1.3.3
        - fsspec:            2024.3.1
        - future:            0.18.3
        - gammatone:         1.0
        - gdown:             4.7.1
        - gitdb:             4.0.10
        - gitpython:         3.1.31
        - gmpy2:             2.1.2
        - gpustat:           1.1
        - gradnorm-pytorch:  0.0.26
        - h11:               0.14.0
        - h5py:              3.11.0
        - httpcore:          0.16.3
        - httptools:         0.5.0
        - httpx:             0.23.3
        - huggingface-hub:   0.22.2
        - human-readable-ids: 0.1.3
        - humanfriendly:     10.0
        - hydra-core:        1.1.1
        - hydra-submitit-launcher: 1.2.0
        - idna:              3.4
        - importlib-metadata: 6.1.0
        - importlib-resources: 5.12.0
        - inquirer:          3.1.3
        - ipykernel:         6.15.0
        - ipython:           8.12.0
        - ipywidgets:        8.0.6
        - itsdangerous:      2.1.2
        - jedi:              0.18.1
        - jinja2:            3.1.2
        - joblib:            1.2.0
        - jsonargparse:      4.20.0
        - jupyter-client:    8.1.0
        - jupyter-core:      5.3.0
        - jupyterlab-widgets: 3.0.7
        - kiwisolver:        1.4.4
        - lazy-loader:       0.2
        - librosa:           0.10.0.post2
        - lightning:         2.0.6
        - lightning-cloud:   0.5.37
        - lightning-utilities: 0.8.0
        - llvmlite:          0.39.1
        - markdown-it-py:    2.2.0
        - markupsafe:        2.1.1
        - mat73:             0.63
        - matplotlib:        3.7.1
        - matplotlib-inline: 0.1.6
        - mdurl:             0.1.2
        - mkl-fft:           1.3.1
        - mkl-random:        1.2.2
        - mkl-service:       2.4.0
        - mock:              5.0.1
        - mpmath:            1.2.1
        - msgpack:           1.0.5
        - multidict:         6.0.4
        - munkres:           1.1.4
        - mypy-extensions:   0.4.3
        - natsort:           7.1.1
        - nest-asyncio:      1.5.6
        - netcdf4:           1.6.4
        - networkx:          2.8.4
        - neural-compressor: 2.2
        - nose:              1.3.7
        - numba:             0.56.4
        - numexpr:           2.8.4
        - numpy:             1.23.5
        - nvidia-ml-py:      11.525.112
        - omegaconf:         2.1.2
        - onnx:              1.14.0
        - onnxruntime-gpu:   1.14.1
        - opencv-python:     4.8.0.74
        - ordered-set:       4.1.0
        - orjson:            3.8.9
        - packaging:         23.0
        - pandas:            1.5.3
        - parso:             0.8.3
        - pathspec:          0.10.3
        - pathtools:         0.1.2
        - pesq:              0.0.4
        - pexpect:           4.8.0
        - pickleshare:       0.7.5
        - pillow:            9.4.0
        - pip:               23.0.1
        - platformdirs:      2.5.2
        - ply:               3.11
        - pooch:             1.4.0
        - prettytable:       3.8.0
        - prompt-toolkit:    3.0.36
        - protobuf:          3.20.3
        - psutil:            5.9.0
        - ptyprocess:        0.7.0
        - pure-eval:         0.2.2
        - py-cpuinfo:        9.0.0
        - pybind11:          2.13.1
        - pyclarity:         0.2.1
        - pycocotools:       2.0.6
        - pycparser:         2.21
        - pydantic:          1.10.7
        - pyfar:             0.5.3
        - pygments:          2.11.2
        - pyjwt:             2.6.0
        - pyloudnorm:        0.1.1
        - pyopenssl:         23.0.0
        - pyparsing:         3.0.9
        - pypesq:            1.2.4
        - pyqt5-sip:         12.11.0
        - pyroomacoustics:   0.7.5
        - pysepm:            0.1
        - pysocks:           1.7.1
        - pystoi:            0.3.3
        - python-dateutil:   2.8.2
        - python-dotenv:     1.0.0
        - python-editor:     1.0.4
        - python-multipart:  0.0.6
        - pytorch-lightning: 2.0.0
        - pytz:              2022.7
        - pyyaml:            6.0
        - pyzmq:             23.2.0
        - readchar:          4.0.5
        - requests:          2.28.1
        - resampy:           0.4.2
        - rfc3986:           1.5.0
        - rich:              13.3.3
        - safetensors:       0.4.3
        - schema:            0.7.5
        - scikit-learn:      1.2.2
        - scipy:             1.10.1
        - seaborn:           0.12.2
        - sentry-sdk:        1.18.0
        - setproctitle:      1.3.2
        - setuptools:        66.0.0
        - sip:               6.6.2
        - six:               1.16.0
        - smmap:             5.0.0
        - sniffio:           1.3.0
        - sofar:             1.1.1
        - soundfile:         0.12.1
        - soupsieve:         2.4
        - soxr:              0.3.4
        - srmrpy:            1.0
        - stack-data:        0.2.0
        - starlette:         0.27.0
        - starsessions:      1.3.0
        - submitit:          1.4.5
        - sympy:             1.11.1
        - threadpoolctl:     3.1.0
        - toml:              0.10.2
        - tomli:             2.0.1
        - torch:             2.0.1
        - torch-fftconv:     0.1.3
        - torchaudio:        2.0.2
        - torchinfo:         1.7.2
        - torchmetrics:      0.11.4
        - torchvision:       0.15.2
        - tornado:           6.2
        - tqdm:              4.65.0
        - traitlets:         5.7.1
        - triton:            2.0.0
        - typeshed-client:   2.2.0
        - typing-extensions: 4.5.0
        - tzdata:            2023.3
        - ujson:             5.7.0
        - urllib3:           1.26.15
        - uvicorn:           0.21.1
        - uvloop:            0.17.0
        - wandb:             0.17.4
        - watchfiles:        0.19.0
        - wcwidth:           0.2.5
        - websocket-client:  1.5.1
        - websockets:        10.4
        - wheel:             0.38.4
        - widgetsnbextension: 4.0.7
        - wrapt:             1.15.0
        - yarl:              1.8.2
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - ELF
        - processor:         x86_64
        - python:            3.10.11
        - release:           5.4.0-153-generic
        - version:           #170-Ubuntu SMP Fri Jun 16 13:43:31 UTC 2023

</details>

**Comment by awaelchli** [comment]
@phuntast1c This is the code we have:
https://github.com/Lightning-AI/pytorch-lightning/blob/2dc9c3d93300269b36e2cfaf1b0b1b32830d2e84/src/lightning/pytorch/trainer/connectors/checkpoint_connector.py#L79-L80

I couldn't really understand what you mean. For bug reports, please provide the necessary information we ask for in the template, including a code example that demonstrates the issue clearly. Thank you.

**Comment by phuntast1c** [comment]
Sorry for the missing information. For some reason, there was the typo mentioned above in the lightning source I was using, but I cannot find the typo in this repository. Thanks for taking the time!
